src/nfts/NftsInfoCarousel.tsx

```diff
@@ -0,0 +1,367 @@
+  return (
+    <SafeAreaView
+      testID="NftsInfoCarousel/NftsLoadErrorScreen"
+      style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}
+    >
+      <Text>{t('nftInfoCarousel.noNftsFound')}</Text>
+    </SafeAreaView>
+  )
+}
+
+const ThumbnailImagePlaceholder = () => {
+  return (
+    <SkeletonPlaceholder backgroundColor={colors.gray2} highlightColor={colors.white}>
+      <View style={styles.skeletonImageThumbnailPlaceHolder} />
+    </SkeletonPlaceholder>
+  )
+}
+
+const NftThumbnail = ({
+  nft,
+  setActiveNft,
+  activeNft,
+}: {
+  nft: Nft
+  setActiveNft: any
+  activeNft: Nft
+}) => {
+  const [loading, setLoading] = useState(true)
+  return (
+    <Touchable borderless={false} onPress={() => setActiveNft(nft)}>
+      <FastImage
+        style={[
+          styles.nftPreviewImageShared,
+          activeNft.tokenUri === nft.tokenUri
+            ? styles.nftPreviewImageSelected
+            : styles.nftPreviewImageUnSelected,
+        ]}
+        source={{
+          uri: nft.media.find((media) => media.raw === nft.metadata?.image)?.gateway,
+        }}
+        onError={() => {
+          Logger.error(TAG, 'Error loading Nft preview image')
+        }}
+        resizeMode={FastImage.resizeMode.cover}
+        onLoadEnd={() => {
+          setLoading(false)
+        }}
+      >
+        {loading && <ThumbnailImagePlaceholder />}
+      </FastImage>
+    </Touchable>
+  )
+}
+
+export default function NftsInfoCarousel({ route }: Props) {
+  const { nfts } = route.params
+  const [activeNft, setActiveNft] = useState((nfts && nfts[0]) ?? null)
+  const [isLoading, setIsLoading] = useState(true)
+  const [scaledHeight, setScaledHeight] = useState(360)
+  const { t } = useTranslation()
+
+  const android_ripple = {
+    color: colors.gray2,
+    foreground: true,
+    borderless: true,
+  }
+
+  // Some components that require parent state defined
+  const PlatformSpecificTopBarButtons = () => {
+    return (
+      <View style={styles.topBarButtonsContainerWrapper}>
+        <View style={styles.topBarButtonsContainer}>
+          {Platform.OS === 'ios' && (
+            <TopBarIconButton
+              onPress={() => navigateBack()}
+              icon={<BackChevronStatic />}
+              style={styles.button}
+              testID="NftsInfoCarousel/BackButton"
+            />
+          )}
+          {Platform.OS === 'android' && (
+            <Pressable
+              android_ripple={android_ripple}
+              onPress={() => navigateBack()}
+              style={styles.button}
+              testID="NftsInfoCarousel/BackButton"
+            >
+              <BackChevronStatic />
+            </Pressable>
+          )}
+        </View>
+      </View>
+    )
+  }
+
+  const MainImagePlaceholder = () => {
+    return (
+      <SkeletonPlaceholder
+        borderRadius={8}
+        backgroundColor={colors.gray2}
+        highlightColor={colors.white}
+      >
+        <View
+          style={{
```

Comment:
use the stylesheet (there are also other places where we should remove the inline styles)

Reply:
I removed most of the inline styles. The ones I left in https://github.com/valora-inc/wallet/pull/3800/commits/b6fc9b3d175c7155c6c3ad03a4007d42be9107a7 were because I started noticing a flicker on the main image after moving the `width` and `zIndex` to stylesheets 🤔